theseus/geometry/se3.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Union, cast
+
+import torch
+
+import theseus
+import theseus.constants
+
+from .lie_group import LieGroup
+from .point_types import Point3
+from .so3 import SO3
+
+
+class SE3(LieGroup):
+    def __init__(
+        self,
+        x_y_z_quaternion: Optional[torch.Tensor] = None,
```

Comment:
Given that we follow _rotation-then-translation_ convention (like gtsam), should we make this input consistent with the convention.

Reply:
This is consistent with the convention in [`SE2`](https://github.com/facebookresearch/theseus/blob/6a92aff8a924e1bf036118d305a5afb3ff195a86/theseus/geometry/se2.py#L24), which uses `x_y_theta` for initialization. I'm OK with either.

Let me know if you would like to change the convention.

Reply:
SE3 data is rotation-then-translation. @luisenp, @dongjing3309 any thoughts?

Reply:
I don't have a strong preference here. Aesthetically it seems more pleasing to me to be consistent between `SE2` and `SE3` (even if it means refactoring `SE2` to use `theta_x_y`). But we should def give priority to any well-established conventions for working with se2 and se3 objects over this.

Reply:
Created #76 for this, since it also involves SE2 changes.